src/components/menu-item/menu-item.tsx

```diff
@@ -0,0 +1,469 @@
+          event,
+          children: this.subMenuItems,
+          isSubMenuOpen: true
+        });
+        break;
+
+      case "ArrowRight":
+        event.preventDefault();
+        if (
+          (event.target === this.dropDownActionEl || !this.href) &&
+          this.hasSubMenu &&
+          !this.open &&
+          this.layout === "vertical"
+        ) {
+          this.open = true;
+          return;
+        }
+        this.calciteInternalMenuItemKeyEvent.emit({
+          event,
+          children: this.subMenuItems,
+          isSubMenuOpen: this.open && this.hasSubMenu
+        });
+        break;
+    }
+  };
+
+  private selectMenuItem(event: MouseEvent | KeyboardEvent): void {
+    if (event.target !== this.dropDownActionEl) {
+      this.calciteMenuItemSelect.emit();
+    }
+  }
+
+  //--------------------------------------------------------------------------
+  //
+  //  Render Methods
+  //
+  //--------------------------------------------------------------------------
+
+  renderIconElStart(): VNode {
+    return (
+      <calcite-icon
+        class={{
+          [CSS.icon]: true,
+          [CSS.iconStart]: true
+        }}
+        flipRtl={this.iconFlipRtl === "start" || this.iconFlipRtl === "both"}
+        icon={this.iconStart}
+        scale="s"
+      />
+    );
+  }
+
+  renderIconElEnd(): VNode {
+    return (
+      <calcite-icon
+        class={{
+          [CSS.icon]: true,
+          [CSS.iconEnd]: true
+        }}
+        flipRtl={this.iconFlipRtl === "end" || this.iconFlipRtl === "both"}
+        icon={this.iconEnd}
+        scale="s"
+      />
+    );
+  }
+
+  renderBreadcrumbIcon(dir: Direction): VNode {
+    return (
+      <calcite-icon
+        class={{
+          [CSS.icon]: true,
+          [CSS.iconBreadcrumb]: true
+        }}
+        icon={dir === "rtl" ? "chevron-left" : "chevron-right"}
+        scale="s"
+      />
+    );
+  }
+
+  renderDropdownIcon(dir: Direction): VNode {
+    const dirChevron = dir === "rtl" ? "chevron-left" : "chevron-right";
+    return (
+      <calcite-icon
+        class={{
+          [CSS.icon]: true,
+          [CSS.iconDropdown]: true
+        }}
+        icon={
+          this.topLevelMenuLayout === "vertical" || this.isTopLevelItem
+            ? this.open
+              ? "chevron-up"
+              : "chevron-down"
+            : dirChevron
+        }
+        id="render-dropdown-icon"
+        scale="s"
+      />
+    );
+  }
+
+  renderDropdownAction(dir: Direction): VNode {
+    const dirChevron = dir === "rtl" ? "chevron-left" : "chevron-right";
+    return (
+      <calcite-action
+        class={CSS.dropdownAction}
+        icon={
+          this.topLevelMenuLayout === "vertical" || this.isTopLevelItem
+            ? this.open
+              ? "chevron-up"
+              : "chevron-down"
+            : dirChevron
+        }
+        onClick={this.clickHandler}
+        onKeyDown={this.keyDownHandler}
+        text="open-dropdown"
```

Comment:
This label needs a nls string right?

Reply:
updated this to `" "` so that we can utilize the existing nls strings for action.

Reply:
The only existing strings for action are for loading and indicator.

Reply:
yeah right. I misunderstood indicator with chevron. This needs to be translated.

Reply:
how about renaming it to `open-submenu`?

cc: @jcfranco , @geospatialem

Reply:
`expand` / `collapse` conditionally based on state if we want to re-use existing translation from Action Bar?

Reply:
I think "Open" might make sense.

Reply:
i second `open` .